x-pack/plugins/security_solution/server/endpoint/routes/metadata/index.ts

```diff
@@ -106,4 +107,23 @@ export function registerEndpointRoutes(
     },
     getMetadataRequestHandler(endpointAppContext, logger)
   );
+
+  router.get(
+    {
+      path: `/api/endpoint/toAgentID`,
+      validate: { query: schema.object({ ids: schema.string() }) },
```

Comment:
Are you missing `authRequired: true` ++ the access tags? (see other routes above)